src/utils/getSelectedTreeOptionFromTreeOptionValue.ts

```diff
@@ -0,0 +1,28 @@
+import { getSelectedOptionFromOptionValue } from './getSelectedOptionFromOptionValue'
+
+import type { Option, OptionValueType, TreeOption } from '@types_/definitions'
+
+function getLeafOptionsFromTreeOptions<OptionValue extends OptionValueType>(
+  treeOptions: Array<TreeOption<OptionValue>>
+): Array<Option<OptionValue>> {
+  return treeOptions.reduce<Array<Option<OptionValue>>>((leafOptions, option) => {
+    if ('value' in option) {
+      return [...leafOptions, option as Option<OptionValue>]
+    }
+
+    return [...leafOptions, ...getLeafOptionsFromTreeOptions(option.children)]
+  }, [])
+}
+
+/**
+ * Get the matching tree option deepest child from a tree option value in a tree option list.
+ */
+export function getSelectedTreeOptionFromTreeOptionValue<OptionValue extends OptionValueType = string>(
```

Comment:
```suggestion
export function getSelectedOptionFromValueInTree<OptionValue extends OptionValueType = string>(
```
Je trouve ça plus clair, non ?